### PR TITLE
raise error instead of silently fail

### LINKED ISSUE #998: Assess what to do in case of no solutions when forking
https://github.com/trailofbits/manticore/blob/3f499225dbf8b02024eef38d7ea1f25a742208ce/manticore/core/executor.py#L384-L385

### PR DESCRIPTION
**Pull request by mossberg**
fixes #998

### CODE REVIEW

File: manticore/core/executor.py [hunk 1]
```diff
@@ -16,6 +17,10 @@
 # This is the single global manager that will handle all shared memory among workers
 
 
+class ExecutorError(ManticoreError):
```
Comment (feliam) [inline manticore/core/executor.py]: It's not important. Raising is better than logging something.

Do we need it to be a ManticoreError at this level. Maybe ExecutorError is better. And let Manticore deal with executor exceptions?

Comment (mossberg) [reply manticore/core/executor.py]: do you mean not have ExecutorError subclass ManticoreError? i think Manticore can still deal with executor exceptions in any case, but there is something i like about having this hierarchy.

File: manticore/manticore.py [hunk 2]
```diff
@@ -17,6 +17,15 @@
 from elftools.elf.elffile import ELFFile
 from elftools.elf.sections import SymbolTableSection
 
+
+# this is here because of circular import issues, various modules imported below need it
+class ManticoreError(Exception):
```
[the remark this answers is not in the document]
Comment (mossberg) [reply manticore/manticore.py]: this is a great point, i see that in a lot of mature projects.

however, i kind of like declaring exceptions in the module where they are used, just for scoping purposes. we have a lot of custom exceptions (evm.py), and moving them all into a single file could be kind of confusing.

also, does that single file contain error classes AND exceptions classes, or just error classes?

[the remark this answers is not in the document]
Comment (disconnect3d) [reply manticore/manticore.py]: > i kind of like declaring exceptions in the module where they are used

Me too, but I found out that when project gets bigger and bigger it is sometimes good to place them in one place.

> also, does that single file contain error classes AND exceptions classes, or just error classes?

To be honest I have no preference, I'd probably be up for having just one file called exceptions (as errors feel like 'worse exceptions').

[the remark this answers is not in the document]
Comment (mossberg) [reply manticore/manticore.py]: i like it. i will work on a separate pr to test it out

**Pull request merged**